pkg/restore/restore.go

```diff
@@ -1260,41 +1318,26 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		if !equality.Semantic.DeepEqual(fromCluster, obj) {
 			switch groupResource {
 			case kuberesource.ServiceAccounts:
-				desired, err := mergeServiceAccounts(fromCluster, obj)
-				if err != nil {
-					ctx.log.Infof("error merging secrets for ServiceAccount %s: %v", kube.NamespaceAndName(obj), err)
-					warnings.Add(namespace, err)
-					return warnings, errs
-				}
-
-				patchBytes, err := generatePatch(fromCluster, desired)
-				if err != nil {
-					ctx.log.Infof("error generating patch for ServiceAccount %s: %v", kube.NamespaceAndName(obj), err)
-					warnings.Add(namespace, err)
-					return warnings, errs
-				}
-
-				if patchBytes == nil {
-					// In-cluster and desired state are the same, so move on to
-					// the next item.
-					return warnings, errs
-				}
-
-				_, err = resourceClient.Patch(name, patchBytes)
-				if err != nil {
-					warnings.Add(namespace, err)
+				return ctx.patchServiceAccount(resourceClient, namespace, name, obj, fromCluster)
+			default:
+				if namespace != "" {
+					// namespaced resource
+					if e := ctx.deleteObject(resourceClient, groupResource, namespace, name); e != nil {
+						warnings.Add(namespace, errors.Wrap(e, "skipping due to deletion failure"))
+						return warnings, errs
+					}
+					createdObj, restoreErr = resourceClient.Create(obj)
 				} else {
-					ctx.log.Infof("ServiceAccount %s successfully updated", kube.NamespaceAndName(obj))
+					// cluster scoped resource
+					e := errors.Errorf("could not restore, %s. Warning: the in-cluster version is different than the backed-up version.", restoreErr)
+					warnings.Add(namespace, e)
```

Comment:
Do the error messages have to start with small letters? This one doesn't wrap right?